invenio_cli/utils.py

```diff
@@ -53,11 +62,84 @@ def destroy_containers(dev):
         subprocess.call(command)
 
 
-def cookiecutter_repo(flavor):
-    """Utility function. Returns the Cookiecutter repository of a flavour."""
-    if flavor.upper() == 'RDM':
-        return {
-                'template': 'https://github.com/inveniosoftware/' +
-                            'cookiecutter-invenio-rdm.git',
-                'checkout': 'v1.0.0a2'
-            }
+class Cookiecutter(object):
+    """Cookiecutter helper object for InvenioCLI."""
+
+    def __init__(self):
+        """Cookiecutter helper constructor."""
+        self.tmp_file = None
+        self.template = None
+
+    def repository(self, flavor):
+        """Get the cookiecutter repository of a flavour."""
+        if flavor.upper() == 'RDM':
+            repo = {
+                    'template': 'https://github.com/inveniosoftware/' +
+                                'cookiecutter-invenio-rdm.git',
+                    'checkout': 'v1.0.0a2'
+                }
+            self.template = 'cookiecutter-invenio-rdm.json'
+            return repo
+
+    def create_and_dump_config(self):
+        """Create a tmp file to store cookicutters used configuration."""
+        if not self.tmp_file:
+            self.tmp_file = \
+                tempfile.NamedTemporaryFile(mode='w+', delete=False)
+
+        config = DEFAULT_CONFIG.copy()
+        # Bug when dumping default {} it's read a string
+        config['default_context'] = None
+        config['replay_dir'] = tempfile.gettempdir()
+
+        yaml.dump(config, self.tmp_file)
```

Comment:
The problem is the following:

1. When running cookiecutter programatically, we only get back the created directory path (string)
2. We want to get all users input, since having a proper context will help us in the upgrade.
3. The only way to get the input is through the *replay* file. Which by default is saved in the user defined cookiecutter directory (which we do not know for sure)
4. In order to change the replay dir, the only way to do so is by creating a *config file* that will be read by cookiecutter. And this config file has to be YAML :man_shrugging:

It's a bit messy but is the only way for now. Until the new Cookiecutter context (https://github.com/cookiecutter/cookiecutter/pull/1008)

Reply:
Ah, so cookiecutter module wants a yaml file... :man_shrugging: indeed.